src/System.Net.Sockets.Legacy/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -2304,6 +2305,7 @@ private void ConnectClient(object state)
                 }
                 catch (SocketException) 
                 {
+                    Task.Delay(100).Wait(); // Give the other end a chance to call Accept().
```

Comment:
I feel the same way, but I don't really see a way to do that in this case. Both threads are going to perform blocking I/O, and releasing the connector immediately before the call to Accept() doesn't necessarily prevent the race (that said, neither does the delay, but I'm a bit more confident in it, as it will involve a yield).